src/Config/RuntimeConfigLoader.cs

```diff
@@ -71,7 +71,24 @@ public static bool TryParseConfig(string json, [NotNullWhen(true)] out RuntimeCo
                 updatedConnectionString = GetConnectionStringWithApplicationName(updatedConnectionString);
             }
 
-            config = config with { DataSource = config.DataSource with { ConnectionString = updatedConnectionString } };
+            if (dataSourceName == null || string.Equals(dataSourceName, config.DefaultDataSourceName, StringComparison.OrdinalIgnoreCase))
```

Comment:
```suggestion
            if (dataSourceName is null null || string.Equals(dataSourceName, config.DefaultDataSourceName, StringComparison.OrdinalIgnoreCase))
```

Keeping consistent with our other code.